tests/data/dataset_readers/wikitables/world_representation_test.py

```diff
@@ -0,0 +1,50 @@
+# pylint: disable=no-self-use,invalid-name
+from allennlp.data.dataset_readers.wikitables import TableKnowledgeGraph, World
+from allennlp.common.testing import AllenNlpTestCase
+
+
+class TestWorldRepresentation(AllenNlpTestCase):
+    def test_world_processes_sempre_forms_correctly(self):
+        table_kg = TableKnowledgeGraph.read_from_file("tests/fixtures/data/wikitables/sample_table.tsv")
+        world = World(table_kg)
+        sempre_form = "((reverse fb:row.row.year) (fb:row.row.league fb:cell.usl_a_league))"
+        expression = world.process_sempre_forms([sempre_form])[0]
+        assert str(expression) == "R(C0,C1(cell:usl_a_league))"
+
+    def test_world_returns_correct_actions_with_reverse(self):
+        table_kg = TableKnowledgeGraph.read_from_file("tests/fixtures/data/wikitables/sample_table.tsv")
+        world = World(table_kg)
+        sempre_form = "((reverse fb:row.row.year) (fb:row.row.league fb:cell.usl_a_league))"
+        expression = world.process_sempre_forms([sempre_form])[0]
+        actions = world.get_action_sequence(expression)
+        target_action_sequence = ['e -> [<r,e>, r]', '<r,e> -> [<<#1,#2>,<#2,#1>>, <e,r>]',
```

Comment:
Shouldn't the first action be the type of the overall expression?

Reply:
Hmm.. yeah. That is a good idea. I'll add that to the action sequence.